### ISSUE TITLE
Position resets / doesn't save.

### ISSUE THREAD

**Issue by ghost** [minetest/minetest#4718]
I imagine this is in duplicate, and I apologize for being vague, but I feel it's worth posting.

The player's position isn't saving in the player's file when using minetestserver, leaving the server running, and joining and disconnecting and re-joining.  Examining the worlddir/players/playername file shows missing or incorrect values.

**Comment by everamzah** [comment]
![pos](https://cloud.githubusercontent.com/assets/1761623/19956579/96a221da-a164-11e6-913f-39c48558015d.png)

Shows in worlddir/players/playername as:
```
breath = 11
hp = 20
name = everamzah
pitch = 8.19
position = (-3456.02,140.6,2793.44)
version = 1
yaw = -44.88
PlayerArgsEnd
```

**Comment by everamzah** [comment]
Other times showing just:
```
name = everamzah
version = 1
PlayerArgsEnd
```

The result is that the player is positioned at 0,0,0 on join.

**Comment by paramat** [comment]
Fixer writes:
///////////////////

When you login, you start at 0,0,0 (not even on spawn or home), your previous position on server is lost for unknown reason.
Happens on Liberty Land (few days old 0.4.14-dev) and Xanadu (0.4.14) to some players... like me.

This bug is different from https://github.com/minetest/minetest/issues/4403 :trollface:

LL admin said to me that roughly 10 days ago build was working OK, soooo my suspects are >_>:
https://github.com/minetest/minetest/commit/9d25242c5c1411d692254cf910345d51c9a24fa3
https://github.com/minetest/minetest/commit/595932a8602292f28333ce14e20cee4b6d8820c1

@nerzhul

**Comment by paramat** [comment]
Possibly related, a bug with 'pitch' was reported by someone reporting this spawn bug, the refactor affected 'pitch'. Error message:
```
2016-11-04 23:21:19: ERROR[Server]: An unhandled exception occurred: Setting [pitch] not found.
2016-11-04 23:21:19: ERROR[Server]: In thread 7fb291584700:
2016-11-04 23:21:19: ERROR[Server]: /home/minetest/minetest/src/server.cpp:73: virtual void* ServerThread::run(): A fatal error occured: Setting [pitch] not found.
2016-11-04 23:21:19: ERROR[Server]: Debug stacks:
2016-11-04 23:21:19: ERROR[Server]: DEBUG STACK FOR THREAD 7fb289efd700: 
2016-11-04 23:21:19: ERROR[Server]: #0  virtual void* EmergeThread::run()
2016-11-04 23:21:19: ERROR[Server]: DEBUG STACK FOR THREAD 7fb28bfff700: 
2016-11-04 23:21:19: ERROR[Server]: #0  virtual void* EmergeThread::run()
2016-11-04 23:21:19: ERROR[Server]: DEBUG STACK FOR THREAD 7fb290d83700: 
2016-11-04 23:21:19: ERROR[Server]: #0  virtual void* CurlFetchThread::run()
2016-11-04 23:21:19: ERROR[Server]: DEBUG STACK FOR THREAD 7fb291584700: 
2016-11-04 23:21:19: ERROR[Server]: #0  virtual void* ServerThread::run()
2016-11-04 23:21:19: ERROR[Server]: DEBUG STACK FOR THREAD 7fb297517740: 
2016-11-04 23:21:19: ERROR[Server]: #0  int main(int, char**)
2016-11-04 23:21:19: ERROR[Server]: #1  Dedicated server branch
2016-11-04 23:21:19: ERROR[Server]: #2  void dedicated_server_loop(Server&, bool&)
```

**Comment by paramat** [comment]
This bug confirmed by bisection as https://github.com/minetest/minetest/commit/9d25242c5c1411d692254cf910345d51c9a24fa3 @nerzhul

**Comment by Zeno-** [comment]
git bisect was done using current head (6bb43472) as bad and c071efaa43 as good

**Comment by paramat** [comment]
I assume thse were the starting commits and good and bad were narrowed down to one either side of 9d25.

**Comment by nerzhul** [comment]
@ghost the file output you show is very strange, it seems at a moment we save player without a player SAO. In proper disconnection this should not happen because the SAO is there. Is there any other reason to save player without SAO ? I should look at improper disconnection, maybe the SAO is removed without any save and this trigger the removal of the player file saved attributes (thanks for file backend)

**Comment by paramat** [comment]
Possible fix #4734

**Comment by nerzhul** [comment]
Fixed by #4734

**Comment by Fixer-007** [comment]
Yes, fixed.
